blockly-core/core/ui/contract_editor/x_button.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+goog.provide('Blockly.XButton');
+
+/**
+ * A DOM-based input which sends events on changes
+ * @param {!Object} options
+ *          onNameChanged (newName: String)
+ *          onEnterPressed
+ * @constructor
+ */
+Blockly.XButton = function (options) {
```

Comment:
This should really be made a "text button", and actually the "Add" button used at the top-level contract editor could then be an instance of this. I'll track this